goagen/gen_client/generator.go

```diff
@@ -664,19 +677,24 @@ func (c *Client) {{ $funcName }}(ctx context.Context, path string{{ if .Payload
 {{ if $headers }}{{ range $name, $att := $params.Type.ToObject }}{{ if (eq $att.Type.Kind 4) }}	header.Set("{{ $name }}", {{ goify $name false }})
 {{ else }}{{ $tmp := tempvar }}{{ toString (goify $name false) $tmp $att }}
 	header.Set("{{ $name }}", {{ $tmp }})
-{{ end }}{{ end }}{{ end }}	header.Set("Content-Type", "application/json")
+{{ end }}{{ end }}{{ end }}	header.Set("Content-Type", "application/json"){{ if .Security }}
+    c.Signer{{ .Security.Scheme.SchemeName }}.Sign(ctx, req){{ end }}
 	return c.Client.Do(ctx, req)
 }
 `
 
 const clientTmpl = `// Client is the {{ .Name }} service client.
 type Client struct {
-	*goaclient.Client
+	*goaclient.Client{{range $security := .SecuritySchemes }}
+    Signer{{ $security.SchemeName }} goaclient.Signer{{ end }}
 }
 
 // New instantiates the client.
 func New(c *http.Client) *Client {
-	return &Client{Client: goaclient.New(c)}
+	return &Client{
+        Client: goaclient.New(c),{{range $security := .SecuritySchemes }}
+        Signer{{ $security.SchemeName }}: &{{ signerType ($security) }}{},{{ end }}
```

Comment:
Please use `Goify` to sanitize the name (may not be a valid Go identifier name). Something like:

 ```go
Signer{{ goify $security.SchemeName true }}: ...
```

Also need to update where it's called (in `RegisterFlags`).